mod-source-record-manager-server/src/main/java/org/folio/dao/JobExecutionDaoImpl.java

```diff
@@ -451,4 +460,64 @@ private String buildOrderByClause(List<SortField> sortFields) {
       .collect(Collectors.joining(", ", "ORDER BY ", EMPTY));
   }
 
+  @Override
+  public Future<RowSet<Row>> hardDeleteJobExecutions(String tenantName, long diffNumberOfDays) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    Promise<SQLConnection> connection = Promise.promise();
+    PostgresClient postgresClient = pgClientFactory.createInstance(tenantName);
+    String rollbackMessage = "Rollback transaction. Error during jobExecution deletion process.";
+    Future.succeededFuture()
+      .compose(c -> {
+        postgresClient.startTx(connection);
+        return connection.future();
+      })
+      .compose(s -> fetchJobExecutionIdsToBeConsideredForDeleting(tenantName, diffNumberOfDays, connection, postgresClient))
+      .compose(rowSet -> {
+        if (rowSet.rowCount() < 1) {
+          throw new NotFoundException(rollbackMessage);
```

Comment:
I think it is possible for this process to run and not find any jobs to hard delete. What do you think?

Reply:
Yes it is possible, approach has been changed without throwing exception(now we use utility class to wrap into transaction, so we could not fail future or throw exception, because it will be considered as action to rollback transaction by this utility. So now we rely on CollectionUtils.isEmpty() checking.